sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/AnalyticsTransmissionTarget.java

```diff
@@ -172,14 +172,34 @@ public void trackEvent(String name) {
      * @param properties Optional properties.
      */
     public void trackEvent(String name, Map<String, String> properties) {
+        trackEvent(name, properties, Flags.DEFAULTS);
+    }
+
+    /**
+     * Track a custom event with name and optional string properties.
+     * <p>
+     * The following rules apply:
+     * <ul>
+     * <li>The event name needs to match the <tt>[a-zA-Z0-9]((\.(?!(\.|$)))|[_a-zA-Z0-9]){3,99}</tt> regular expression.</li>
+     * <li>The property names or values cannot be null.</li>
+     * <li>The <tt>baseData</tt> and <tt>baseDataType</tt> properties are reserved and thus discarded.</li>
+     * <li>The full event size when encoded as a JSON string cannot be larger than 1.9MB.</li>
+     * </ul>
+     *
+     * @param name       An event name.
+     * @param properties Optional properties.
+     * @param flags      Optional flags. Use {@link Flags#PERSISTENCE_CRITICAL} to send this event
+     *                   before events using that use default flags or {@link Flags#PERSISTENCE_NORMAL}.
```

Comment:
wording can be improved, maybe something like:
```Use {@link Flags#PERSISTENCE_CRITICAL} to send this event before other events with {@link Flags#PERSISTENCE_NORMAL} or default flags.```

Reply:
Let's just align with Apple then: https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1208#discussion_r231738233